airflow/providers/amazon/aws/triggers/redshift_data.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any, AsyncIterator
+
+from airflow.providers.amazon.aws.hooks.redshift_data import RedshiftDataHook
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class RedshiftDataTrigger(BaseTrigger):
+    """
+    RedshiftDataTrigger is fired as deferred class with params to run the task in triggerer.
+
+    :param statement_id: the UUID of the statement
+    :param task_id: task ID of the Dag
+    :param poll_interval:  polling period in seconds to check for the status
+    :param aws_conn_id: AWS connection ID for redshift
+    :param region: aws region to use
+    """
+
+    def __init__(
+        self,
+        statement_id: str,
+        task_id: str,
+        poll_interval: int,
+        aws_conn_id: str | None = "aws_default",
+        region: str | None = None,
```

Comment:
At that moment there is quite a few inconsistencies exist in Amazon related operators/sensors/triggers. One of them is `region` vs `region_name`. There is some changes already made in operators, however there is still some of them use `region`, but other already deprecate `region` in favor of the `region_name` (this name use in Hooks).

So I would recommend rename this argument to `region_name` so it would use same naming

```suggestion
        region_name: str | None = None,
```

In additional I also would like to recommend add two parameters `verify` and `botocore_config` with default values as None, so that make hook parameters consistent with parameters which propagated from Operator to `RedshiftDataHook`, as reference you could use [`SqsSensorTrigger`](https://github.com/apache/airflow/blob/81be6ac6c091c9c922c4060b12fc92e2d7ad424d/airflow/providers/amazon/aws/triggers/sqs.py#L31)

Reply:
Hi @Taragolis, Thanks for your suggestion! I just applied them to this PR